client/pd_service_discovery.go

```diff
@@ -404,10 +371,6 @@ func (c *pdBaseClient) updateMember() error {
 		if err := c.switchLeader(members.GetLeader().GetClientUrls()); err != nil {
 			return err
 		}
-		// Run callbacks to refelect any change in the local/global tso allocator.
-		for _, cb := range c.tsoAllocatorLeaderSwitchedCallback {
-			cb()
-		}
```

Comment:
no need to run callbacks now?

Reply:
It still needs to run callbacks. In c.switchTSOAllocatorLeader() above, now it will invoke c.tsoAllocLeadersUpdatedCbs, i.e., invoke tsoClient.updateTSOAllocatorServingAddrs(). This actually fixed a bug in the legacy code, when c.switchTSOAllocatorLeader switched local allocator leaders but c.switchLeader returns error, previously we didn't call c.tsoAllocatorLeaderSwitchedCallback, which isn't correct.